doajtest/mocks/models_Cache.py

```diff
@@ -26,6 +26,16 @@ def cache_sitemap(cls, filename):
             "filename" : filename
         }
 
+    @classmethod
+    def cache_nth_sitemap(cls, n, url):
+        cls.__memory__["sitemap" + str(n)] = {
+            "filename": url
+        }
+
+    @classmethod
+    def get_sitemap(cls, n):
+        return cls.__memory__["sitemap" + str(n)]
```

Comment:
When I run the unit tests I get 2 failures, and they are because `models.Cache.pull` is used in the main code to access the sitemap cache for pruning, but the `pull` method does not exist here in the mock cache.

I'm not sure what the right solution is: either add `pull` to the mock cache, or have a specific method for retrieving the sitemap cache record that is used in the main code.

Reply:
Added pull method to InMemoryCache as the pull method is used at multiple places.